app/models/nextUpdates/NextUpdatesResponseModel.scala

```diff
@@ -42,6 +43,10 @@ case class ObligationsModel(obligations: Seq[NextUpdatesModel]) extends NextUpda
     if (previous) deadlines.sortBy(_.obligation.dateReceived.map(_.toEpochDay)).reverse else deadlines.sortBy(_.obligation.due.toEpochDay)
   }
 
+  def getPropertyType(model: NextUpdatesModel)(implicit mtdItUser: MtdItUser[_]) = {
+    val incomeSource = mtdItUser.incomeSources.properties.filter(_.incomeSourceId == model.identification)
```

Comment:
* we are duplicating logic in the scope of the single method(s), I don't think method 'getPropertyType' is required:
consider next approach:

```
 val deadlines = obligations.flatMap { deadlinesModel =>
      mtdItUser.incomeSources.properties.find(_.incomeSourceId == deadlinesModel.identification) match {
        case Some(property) if property.incomeSourceType.contains("foreign-property") =>
          deadlinesModel.obligations.map {
            deadline => Some(NextUpdateModelWithIncomeType(s"nextUpdates.propertyIncome.${getPropertyType(deadlinesModel)}", deadline))
          }
        case Some(property)  => // assume this is a UK property
            deadlinesModel.obligations.map {
              deadline => Some(NextUpdateModelWithIncomeType(s"nextUpdates.propertyIncome.${getPropertyType(deadlinesModel)}", deadline))
            }
        case None => //  
      }
```